src/adapters/amqp-node.ts

```diff
@@ -24,61 +20,33 @@ export interface QueueOptions {
   }
 }
 
-export interface ExchangeOptions {
-  durable: boolean;
-}
-
-interface ConsumeResult {
-  consumerTag: string;
-}
-
-export interface AMQPConnection {
-  assertQueue: (queue: string, options: QueueOptions) => Promise<void>;
-  assertExchange: (exchange: string, queue: string, options: ExchangeOptions) => Promise<void>;
-  ack: (message: Message) => void;
-  nack: (message: Message) => void;
-  sendToQueue: (queue: string, message: Buffer, options: MessageOptions) => Promise<void>;
-  bindQueue: (queue: string, exchange: string, routingRegExp: string) => Promise<void>;
-  publish: (exchange: string, routingKey: string, message: Buffer, options: MessageOptions) => Promise<void>;
-  consume: (queue: string, handler: (message: RawMessage) => void) => Promise<ConsumeResult>;
-  prefetch: (maxMessages: number) => Promise<void>;
-  cancel: (consumerTag: string) => Promise<void>;
-  close: () => Promise<void>;
-}
+export type AMQPConnection = Pick<AMQPChannel, 'basicAck' | 'basicCancel' | 'basicConsume' | 'basicNack' | 'basicPublish' | 'close' | 'exchangeDeclare' | 'prefetch' | 'queue' | 'queueBind'>;
```

Comment:
It is safer to declare own interfaces in mbclient rather than blindly reexport from amqp-client.

That way allows us explicitly control types and handle their mismatch immediately.

Reply:
Reexported only used amp-client.js type subset.